### ISSUE TITLE
How to train your model？

### ISSUE THREAD

**Issue by hescluke** [Hongje/HMMN#6]
I noticed that this code only included evaluation part and lack the training part. It would be really helpful if you could proide the training code. Another question is that the model in this code has some differences compared to the paper. In the top-k guided memory matching module, the model in this code don't have the VQ3 add to the result which come out from dropout. Instead, it directly uses results after the second ''matmul'' as the output of the top-k module (in Figure.6 in the paper this is Z4). Could you please explain the reason? Moreover, I wonder if I need to add a dropout layer, a conv3 layer and a ''VQ3 add.'' layer when training the model. Of course if you can provide me with your training  code, it would be most helpful! Thank you for your time!
![6650b00feaa8005421926ba4506cbe5](https://user-images.githubusercontent.com/86816786/184587269-8be40dc3-28b5-42f5-8f57-442a83b45baf.jpg)
